spynnaker/pyNN/models/utility_models/synapse_expander/synapse_expander.py

```diff
@@ -61,34 +65,14 @@ def synapse_expander(
     expander_cores, expanded_pop_vertices = _plan_expansion(
         app_graph, placements, synapse_bin, delay_bin, progress)
 
-    # Launch the delay receivers
     expander_app_id = transceiver.app_id_tracker.get_new_id()
-    transceiver.execute_application(expander_cores, expander_app_id)
-    progress.update()
-
-    # Wait for everything to finish
-    finished = False
-    try:
-        transceiver.wait_for_cores_to_be_in_state(
-            expander_cores.all_core_subsets, expander_app_id,
-            [CPUState.FINISHED])
-        progress.update()
-        finished = True
-        _fill_in_connection_data(
-            expanded_pop_vertices, placements, transceiver)
-        _extract_iobuf(expander_cores, transceiver, provenance_file_path)
-        progress.end()
-    except Exception:  # pylint: disable=broad-except
-        logger.exception("Synapse expander has failed")
-        _handle_failure(
-            expander_cores, transceiver, provenance_file_path)
-    finally:
-        transceiver.stop_application(expander_app_id)
-        transceiver.app_id_tracker.free_id(expander_app_id)
-
-        if not finished:
-            raise SpynnakerException(
-                "The synapse expander failed to complete")
+    run_system_application(
+        expander_cores, expander_app_id, transceiver, provenance_file_path,
+        executable_finder, extract_iobuf, functools.partial(
+            _fill_in_connection_data, placements=placements,
+            expanded_pop_vertices=expanded_pop_vertices),
+        None, [CPUState.FINISHED], False, no_sync_changes,
```

Comment:
Since needs_sync_barrier is False, no_sync_changes can be ignored.

Reply:
so ignoring the fact your griping about a param from a FEC call. which is beyond this pr (and this pr is bust)

just so we're clear, your saying no_sync_changes should default to None in the FEC call, so that it can handle the False better?

Reply:
Given the bug report, it shouldn't be there at all...  From the perspective of this PR, there should be no "no_sync_changes" argument to the function at all, as it will be ignored inside anyway.

Reply:
> Just realised that there might be an issue here: the new parameter hasn't been added to the XML (or the config file).

did say the pr was broken. i noticed that and was about to fix it till other things distracted me. 

teach me for rushing this coz i WAS VERY HUNGRY!